cdap-ui/app/cdap/components/DataPrep/Directives/ExtractFields/UsingPatternsModal/index.js

```diff
@@ -0,0 +1,432 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import React, {Component, PropTypes} from 'react';
+import {Modal, ModalHeader, ModalBody, ModalFooter} from 'reactstrap';
+import T from 'i18n-react';
+import isNil from 'lodash/isNil';
+import isEmpty from 'lodash/isEmpty';
+import {Input, DropdownToggle, DropdownMenu, DropdownItem} from 'reactstrap';
+import {execute} from 'components/DataPrep/store/DataPrepActionCreator';
+const PREFIX = 'features.DataPrep.Directives.ExtractFields.UsingPatterns';
+import MouseTrap from 'mousetrap';
+import DataPrepStore from 'components/DataPrep/store';
+import DataPrepActions from 'components/DataPrep/store/DataPrepActions';
+import {UncontrolledDropdown} from 'components/UncontrolledComponents';
+
+require('./UsingPatternsModal.scss');
+
+export default class UsingPatternsModal extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      pattern: null,
+      patternName: null,
+      patternLabel: null,
+      startAndEndPattern: {
+        start: '',
+        end: ''
+      },
+      nDigitPattern: '',
+      error: '',
+      showEditPatternLabel: false,
+      showEditPatternTxtBox: false
+    };
+    this.patterns = [
```

Comment:
Talked to @elfenheart. Right now the patterns have mostly just simple string values but some also have `getRegex` function which needs to be evaluated based on user input which will update the actual pattern. For now lets have it as plain array and tomorrow when we add more regex patterns and if it becomes a hassle will extract this out to make it more easier.